### PR TITLE
Feature/tt 4295

### PR DESCRIPTION
**Pull request by sarahentzel**

### CODE REVIEW

File: src/crud/updateBackTranslationType.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  token: string | null,
+  user: string,
+  errorReporter: any
+) => {
+  if (token) {
+    await axiosGet('mediafiles/wbt', undefined, token)
+      // eslint-disable-next-line no-loop-func
+      .then((response) => {
+        var fr = response.data as any;
+        console.log(fr);
```
Comment (gtryus) [inline src/crud/updateBackTranslationType.ts:20]: It looks like all we do with the response data is put it on the console log? Do we need a comment or more code here?

File: src/crud/updateBackTranslationType.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    var artifacttypes = memory.cache.query((q: QueryBuilder) =>
+      q.findRecords('artifacttype')
+    ) as ArtifactType[];
+    var bt =
+      artifacttypes.filter((a) => a.attributes.typename === 'backtranslation')
+        .length > 0
+        ? artifacttypes.filter(
+            (a) => a.attributes.typename === 'backtranslation'
+          )[0]
+        : null;
```
Comment (gtryus) [inline src/crud/updateBackTranslationType.ts:42]: Would artifacttypes.find((a) => a.attributes.typename === 'backtranslation'); not be better? I think it returns the record or undefined.

File: src/crud/updateBackTranslationType.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    var wbt =
+      artifacttypes.filter(
+        (a) => a.attributes.typename === 'wholebacktranslation'
+      ).length > 0
+        ? artifacttypes.filter(
+            (a) => a.attributes.typename === 'wholebacktranslation'
+          )[0]
+        : null;
```
Comment (gtryus) [inline src/crud/updateBackTranslationType.ts:51]: see comment on lines 32-42

File: src/crud/updateBackTranslationType.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        .findRecords('mediafile')
+        .filter({ relation: 'artifactType', record: bt })
+        .filter({ attribute: 'sourceSegments', value: null })
+    ) as MediaFile[];
+
+    mediafiles.forEach((m) =>
+      memory.update((t: TransformBuilder) =>
+        UpdateRelatedRecord(t, m, 'artifactType', 'artifacttype', wbt?.id, user)
+      )
+    );
```
Comment (gtryus) [inline src/crud/updateBackTranslationType.ts:66]: alternatively:
for (const m of mediafiles) {
   await memory.update((t: TransformBuilder) =>
       UpdateRelatedRecord(t, m, 'artifactType', 'artifacttype', wbt?.id, user);
}
which will have the updates done in order ans synchronously. (The {} aren't needed if you don't want them.)

**Review by gtryus — COMMENTED** [review]
Sara, I made some comments for your consideration but I am happy if after you review the comments you check in what you think is best. If you want me to look again, I can. I am most curious about updateBackTranslationType.ts line 28?

File: src/crud/updateBackTranslationType.ts (hunk 4 above)
Comment (sarahentzel) [reply src/crud/updateBackTranslationType.ts:66]: I don't care about the order...and since this is offline only, async is fast enough

**Pull request merged**